client/state/data-layer/wpcom/users/auth-options/index.js

```diff
@@ -0,0 +1,66 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import {
+	LOGIN_AUTH_ACCOUNT_TYPE_REQUESTING,
+	LOGIN_AUTH_ACCOUNT_TYPE_REQUEST_SUCCESS,
+	LOGIN_AUTH_ACCOUNT_TYPE_REQUEST_FAILURE,
+} from 'state/action-types';
+import { noRetry } from 'state/data-layer/wpcom-http/pipeline/retry-on-failure/policies';
+
+export const getAuthAccountType = ( { dispatch }, action ) => {
+	const { usernameOrEmail } = action;
+
+	dispatch(
+		http(
+			{
+				path: `/users/${ usernameOrEmail }/auth-options`,
+				method: 'GET',
+				apiVersion: '1.1',
+				retryPolicy: noRetry(),
+			},
+			action
+		)
+	);
+};
+
+export const receiveSuccess = ( { dispatch }, action, data ) => {
+	const isPasswordless = get( data, 'passwordless' );
+
+	dispatch( {
+		type: LOGIN_AUTH_ACCOUNT_TYPE_REQUEST_SUCCESS,
+		data: {
+			type: isPasswordless ? 'passwordless' : 'regular',
+		},
+	} );
+};
+
+export const receiveError = ( { dispatch }, action, error ) => {
+	dispatch( {
+		type: LOGIN_AUTH_ACCOUNT_TYPE_REQUEST_FAILURE,
+		error: {
+			code: error.error,
+			message: error.message,
+			field: 'usernameOrEmail',
+		},
+	} );
+};
+
+const getAuthAccountTypeRequest = dispatchRequest(
+	getAuthAccountType,
+	receiveSuccess,
+	receiveError,
+);
+
+export default {
+	[ LOGIN_AUTH_ACCOUNT_TYPE_REQUESTING ]: [ getAuthAccountTypeRequest ],
```

Comment:
the convention in the codebase seems to be to have a noun here, `LOGIN_AUTH_ACCOUNT_TYPE_REQUEST`

Reply:
The problem in this pull request is that we need to trigger an event when the user submits the login form, and another one when we actually call the API. That's why I actually used both `LOGIN_AUTH_ACCOUNT_TYPE_REQUEST` and `LOGIN_AUTH_ACCOUNT_TYPE_REQUESTING` - a pattern that is used in [other parts of our code](https://github.com/Automattic/wp-calypso/blob/4924ba7c40685cb7370309e65df42b5f1a59ca60/client/state/action-types.js#L410-L411).

Reply:
I'm sorry, I still don't really understand why we need that action, it's not used anywhere at the reducer, the loading state is derived from `LOGIN_AUTH_ACCOUNT_TYPE_REQUEST`.

Why we need it? there is no reducer for it, we can just use `LOGIN_AUTH_ACCOUNT_TYPE_REQUEST`
see: https://github.com/Automattic/wp-calypso/pull/19688/commits/475bbe6033b6ea78f70f863cb5c226bd9ea5d5ee

In the `MEDIA_REQUESTING` case, we can as well do the same, look at `MEDIA_REQUEST` and not `MEDIA_REQUESTING` `client/state/media/reducer` using `MEDIA_REQUESTING` instead using `MEDIA_REQUEST`.